packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardHacks/private/useValueRef.ts

```diff
@@ -0,0 +1,11 @@
+import { useRef } from 'react';
+
+import type { RefObject } from 'react';
+
+export default function useValueRef<T>(value: T): RefObject<T> {
+  const ref = useRef<T>(value);
+
+  ref.current = value;
```

Comment:
should we wrap this is a `useEffect`?

```
useEffect(() => { ref.current = value;});
```

Reply:
Not really. Unlike `usePrevious` hook which change the ref value after render (remember previous value), this hook change the value immediately. So the hook caller can immediately read the new value.

Before using this hook:

```js
const SomeComponent = ({ value }) => {
  const handleClick = useCallback(() => {
    // This function will be recreated every time `value` change. However, this is unnecessary because the `value` is only used post-render (during `onClick` event).
    alert(value);
  }, [value]);

  return <button onClick={handleClick}>Click me</button>;
}
```

After using this hook:

```js
const SomeComponent = ({ value }) => {
  const valueRef = useValueRef(value);
  const handleClick = useCallback(() => {
    alert(valueRef.current)
  }, [valueRef]);

  return <button onClick={handleClick}>Click me</button>;
}
```

I am thinking to create a `useCallbackWithCheck` to assert that we don't recreate callback functions after initial render, so we can detect this type of wasted renderings.